main.cpp

```diff
@@ -62,11 +64,25 @@ std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
   return std::make_unique<CommandLineUI>();
 }
 
-}  // namespace
+std::unique_ptr<GenericUI> CreateUI(int argc, char** argv) {
+  return std::make_unique<WindowsUI>();
+}
+
+const bool Type_of_UI = ReadCommandLine(argc, argv);
 
+}  // namespace
 
 int main(int argc, char** argv) {
   auto ifg = CreateUI(argc, argv);
   ifg->Run();
   return 0;
+}
+//-----------------------------------Functions------------------------------//
+
+const bool Type_of_UI = ReadCommandLine(argc, argv);
```

Comment:
The idea is to make this logic in the CreateUI functio, where you get the command line arguments. Here, outside all scope, it does not even compile.